### PR TITLE
Id harmonization

### PR DESCRIPTION
**Pull request by bouhamedmohamed**
ID harmonization #26 
In this version we 
-Add a CandidateID which will be the id of Candidate and Checkin Class
-Add a SponsorID which will be the id Sponsor Class

### CODE REVIEW

File: src/main/java/fr/socrates/domain/sponsor/Sponsor.java [hunk 1]
```diff
@@ -79,37 +81,38 @@ public SponsorBuilder withAmountOfSponsoring(double amountOfSponsoring) {
         }
 
         public Sponsor createSponsor() {
-            if((siren==null || !this.isSirenValid(siren)) || (siret != null && !isSiretSyntaxValide(siret)))
+            if ((siren == null || !this.isSirenValid(siren))) {
                 throw new IllegalStateException();
```
Comment (udebella) [inline src/main/java/fr/socrates/domain/sponsor/Sponsor.java]: Can you add some explicit message about why the exception was thrown?

File: src/main/java/fr/socrates/domain/sponsor/Sponsor.java [hunk 2]
```diff
@@ -79,37 +81,38 @@ public SponsorBuilder withAmountOfSponsoring(double amountOfSponsoring) {
...
-            return new Sponsor(name, siret, siren, contractRepresentative, contact, amountOfSponsoring);
+            } else if ((siret != null && !isSiretSyntaxValide(siret))) {
+                throw new IllegalStateException();
```
Comment (udebella) [inline src/main/java/fr/socrates/domain/sponsor/Sponsor.java]: Same here

**Review by udebella — CHANGES_REQUESTED** [review]
IllegalStateException should be corrected

File: src/main/java/fr/socrates/domain/sponsor/Sponsor.java (hunk 1 above)
Comment (bouhamedmohamed) [reply src/main/java/fr/socrates/domain/sponsor/Sponsor.java]: done

File: src/main/java/fr/socrates/domain/sponsor/Sponsor.java (hunk 2 above)
Comment (bouhamedmohamed) [reply src/main/java/fr/socrates/domain/sponsor/Sponsor.java]: Done

**Review by udebella — APPROVED** [review]

**Pull request merged**